### PR TITLE
[flutter_tools] handle invalid yaml in plugin dependency

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Do not crash if a dependency has an invalid plugin specification. It was hard to find invalid yaml, so I used: https://stackoverflow.com/questions/40872314/invalid-yaml-file

### CODE REVIEW

File: packages/flutter_tools/lib/src/plugins.dart [hunk 1]
```diff
@@ -311,7 +311,13 @@ Plugin _pluginFromPackage(String name, Uri packageRoot) {
   if (!globals.fs.isFileSync(pubspecPath)) {
     return null;
   }
-  final dynamic pubspec = loadYaml(globals.fs.file(pubspecPath).readAsStringSync());
+  dynamic pubspec;
+
+  try {
+    pubspec = loadYaml(globals.fs.file(pubspecPath).readAsStringSync());
+  } on YamlException {
+    // Do nothing, potentially not a plugin.
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/plugins.dart:319]: I'm not familiar with this code, should we printError here, or is this expected to run into `YamlExceptions`s?

**Review by christopherfujino — APPROVED** [review]
LGTM, with a nit question about if we should printError when catching `YamlException`

File: packages/flutter_tools/lib/src/plugins.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/plugins.dart:319]: At this point we're still discovering plugins - I think it should definitely be non-fatal, but I'm not really sure how actionable an error parsing the yaml would be. I will add it to the trace so it gets surfaced somehow

**Comment by jonahwilliams** [comment]
Updated with trace output

File: packages/flutter_tools/lib/src/plugins.dart (hunk 1 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/plugins.dart:320]: Yeah, trace makes more sense actually

**Pull request merged**